src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(DependencySettings::isSettings)
+                .collect(toList());
+        List<DependencySettings> externalDependencies = dependencies.stream()
+                .filter(DependencySettings::isExternal)
+                .collect(toList());
+        if (externalDependencies == null) {
+            externalDependencies = Collections.emptyList();
+        }
+        if (dependenciesSettings == null) {
+            dependenciesSettings = Collections.emptyList();
+        }
+
+        for (DependencySettings externalDependency : externalDependencies) {
+            if (externalDependency.path.isDirectory() && (externalDependency.pack != null && !PackingType.NONE.userValue.equals(externalDependency.pack))) {
+                throw new JetTaskFailureException(s("JetApi.NotPackedExternalDirectory", externalDependency.path));
+            }
+        }
+
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            List<ProjectDependency> matchedDependencies = dependenciesMatchedBy(dependencySettings);
+            if (dependencySettings.isArtifactOnly() && matchedDependencies.size() > 1) {
+                List<String> dependencyIds = matchedDependencies.stream().
+                        map(ProjectDependency::idStr).
+                        collect(toList());
+                throw new JetTaskFailureException(s("JetApi.AmbiguousArtifactIdOnlyDependencySettings", dependencySettings.idStr(), String.join(", ", dependencyIds)));
+            }
+        }
+
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            if (dependencySettings.groupId == null && dependencySettings.artifactId == null) {
+                throw new JetTaskFailureException(s("JetApi.DependencySettingsWithoutGroupAndArtifact"));
+            }
+        }
+
+        for (DependencySettings externalDependency: externalDependencies) {
+            if (externalDependency.groupId != null || externalDependency.artifactId != null ||
+                    externalDependency.version != null) {
+                throw new JetTaskFailureException(s("JetApi.ExternalDependencyShouldNotHaveId", externalDependency.path));
+            }
+        }
+
+        for (DependencySettings dependencySettings: dependenciesSettings) {
+            if (dependenciesMatchedBy(dependencySettings).size() == 0) {
+                throw new JetTaskFailureException(s("JetApi.NoDependenciesForDependencySettings", dependencySettings.idStr()));
+            }
+        }
+
+        for (DependencySettings dep : dependencies) {
+            if (dep.packagePath != null && dep.disableCopyToPackage != null && dep.disableCopyToPackage) {
+                String msg = dep.isSettings() ?
+                        s("JetApi.DependencySettingsCannotHavePackagePathAndDisabledCopyToPackage", dep.idStr()) :
```

Comment:
I believe we can use the same message here for both external and not external dependencies